satellite/gracefulexit/endpoint.go

```diff
@@ -330,7 +330,39 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 			err = endpoint.handleSucceeded(ctx, pending, nodeID, m)
 			if err != nil {
 				if ErrInvalidArgument.Has(err) {
-					return rpcstatus.Error(rpcstatus.InvalidArgument, err.Error())
+					// immediately fail and complete graceful exit for nodes that fail satellite validation
+					// TODO(moby) use getFinishedMessage helper after 3368 is merged
+					exitStatusRequest := &overlay.ExitStatusRequest{
+						NodeID:         nodeID,
+						ExitFinishedAt: time.Now().UTC(),
+						ExitSuccess:    false,
+					}
+					// TODO needs signature
```

Comment:
won't getFinishedMessage add the signature too?

Reply:
It should